experimental/redux/otelMiddleware.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  Dispatch,
+  Middleware,
+  MiddlewareAPI,
+  Action,
+  UnknownAction,
+} from "redux";
+import {TracerProvider, trace} from "@opentelemetry/api";
+
+interface ConfigArgs {
+  name: string;
+  version: string;
+}
+
+const otelMiddleware = <RootState>(
+  provider: TracerProvider,
+  config: ConfigArgs,
+): Middleware<object, RootState> => {
+  if (!provider) {
+    console.warn("Provider is not available. Spans won't be recorded");
```

Comment:
similar to my comment on the navigation package I think we should be fine always grabbing the global tracer if there's no provider specified, if nothing is setup then the span stuff will all just be no-ops